GEODE-6533: Update LICENSE and NOTICE files for changed dependencies

sboorlagadda opened the pull request:
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

sboorlagadda commented:
**MIT**
   
animal-sniffer-annotations-1.17.jar - MIT - https://mvnrepository.com/artifact/org.codehaus.mojo/animal-sniffer-annotations
checker-qual-2.5.2.jar - MIT - https://mvnrepository.com/artifact/org.checkerframework/checker-qual

**Apache 2.0**

error_prone_annotations-2.2.0.jar - Apache 2.0 - https://mvnrepository.com/artifact/com.google.errorprone/error_prone_annotations
failureaccess-1.0.jar - Apache 2.0 - https://mvnrepository.com/artifact/com.google.guava/failureaccess
j2objc-annotations-1.1.jar - Apache 2.0 - https://mvnrepository.com/artifact/com.google.j2objc/j2objc-annotations
jsr305-3.0.2.jar - Apache 2.0 - https://mvnrepository.com/artifact/com.google.code.findbugs/jsr305
listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar - Apache 2.0 - https://mvnrepository.com/artifact/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava
commons-math3-3.2.jar - Apache 2.0 - https://mvnrepository.com/artifact/org.apache.commons/commons-math3
geo-0.7.1.jar - Apache 2.0 - https://mvnrepository.com/artifact/com.github.davidmoten/geo
grumpy-core-0.2.2.jar - Apache 2.0 - https://mvnrepository.com/artifact/com.github.davidmoten/grumpy-core

**CDDL**

istack-commons-runtime-2.2.jar - CDDL 1.1 -  https://projects.eclipse.org/projects/ee4j.jaxb 
javax.activation-1.2.0.jar - CDDL 1.1 - https://mvnrepository.com/artifact/javax.activation, https://eclipse-ee4j.github.io/jaf/ 
javax.activation-api-1.2.0.jar - CDD 1.1 - https://mvnrepository.com/artifact/javax.activation, https://eclipse-ee4j.github.io/jaf/

dickcav commented on LICENSE:
```diff
@@ -380,3 +384,377 @@ domain:
   - CompactConcurrentHashSet2, derived from JSR-166 ConcurrentHashMap v1.43
     (http://gee.cs.oswego.edu/dl/concurrency-interest).
   - tooltip.js v1.2.6 (https://github.com/jquerytools/jquerytools)
+
```
It's not clear to me whether this CDDL 1.1 License belongs here in the top LICENSE or in the geode-assembly/src/main/dist/LICENSE which already has a CDDL 1.0 license within that gets appended to the top level license during assembly. Do we need both 1.0 and 1.1 for CDDL?

dickcav commented on LICENSE:
```diff
@@ -295,6 +295,10 @@ Apache Geode bundles the following files under the MIT license:
   - Timeago v1.3.0 (http://timeago.yarp.com), Copyright (c) 2008-2015 Ryan
     McGeary
   - zTree v3.5.02 (http://zTree.me/), Copyright (c) 2010 Hunter.z
+  - Animal Sniffer v1.17 (http://www.mojohaus.org/animal-sniffer), Copyright
```
Shouldn't this Animal Sniffer copyright live in the geode-web/src/main/webapp/META-INF/NOTICE or ./geode-web-management/src/main/webapp/META-INF/NOTICE instead of in the top LICENSE file? 
Generally I don't know when one should be used over the other except in this case the use of Animal Sniffer is only in the geode-web and geode-web-management subprojects so this probably belongs there.

dickcav commented on LICENSE:
```diff
@@ -295,6 +295,10 @@ Apache Geode bundles the following files under the MIT license:
   - zTree v3.5.02 (http://zTree.me/), Copyright (c) 2010 Hunter.z
+  - Animal Sniffer v1.17 (http://www.mojohaus.org/animal-sniffer), Copyright
+    (c) 2008–2017 MojoHaus.
+  - Checker Qual v2.7.0 (https://checkerframework.org), Copyright
```
This seems the same as Animal Sniffer as it's used only by the geode-web and geode-web-management subprojects.

sboorlagadda commented:
@dickcav Thanks for catching the LICENSE file update. Looks like we need to update top-level file if we include any dependency into *src* distribution otherwise we update `geode-assembly/src/main/dist/LICENSE` file.
I have corrected the LICENSE changes and moved those into assembly LICENSE file.

Also, I have reverted NOTICE file changes as I still have more work to do, so will create a new PR.

sboorlagadda commented:
@metatype This PR has been updated with only LICENSE changes and changes to NOTICES will follow.

metatype commented:
@sboorlagadda here's my take on the changes to LICENSE:

```
diff --git a/LICENSE b/LICENSE
index a243c40007..0dc431d441 100644
--- a/LICENSE
+++ b/LICENSE
@@ -257,8 +257,6 @@ The MIT License (http://opensource.org/licenses/mit-license.html)

 Apache Geode bundles the following files under the MIT license:

-  - ClassGraph v4.0.6 (https://github.com/classgraph/classgraph), Copyright (c)
-    2015 Luke Hutchison
   - HTML5 Shiv vpre3.5 (https://github.com/aFarkas/html5shiv), Copyright
     (c) 2014 Alexander Farkas (aFarkas)
   - JavaScript InfoVis Toolkit v2.0.1 (http://philogb.github.io/jit/),
diff --git a/geode-assembly/src/main/dist/LICENSE b/geode-assembly/src/main/dist/LICENSE
index 94a70e636f..fa7db6036e 100644
--- a/geode-assembly/src/main/dist/LICENSE
+++ b/geode-assembly/src/main/dist/LICENSE
@@ -221,6 +221,10 @@ Apache Geode bundles the following files under the BSD 3-Clause License:
     Copyright (c) 2002-2007 Marc Prud'hommeaux.
   - Antlr v2.7.7 (http://www.antlr.org), Copyright (c) 2012 Terrence Parr
     and Sam Harwell
+  - Checker Qual 2.5.2 (https://checkerframework.org), Copyright (c)
+    2004-present by the Checker Framework developers
+  - ClassGraph v4.0.6 (https://github.com/classgraph/classgraph), Copyright (c)
+    2015 Luke Hutchison
   - JLine v2.12 (http://jline.sourceforge.net), Copyright (c) 2002-2006,
     Marc Prud'hommeaux <mwp1@cornell.edu>
   - jQuery Sparklines v2.0 (http://omnipotent.net/jquery.sparkline/),
@@ -270,13 +274,15 @@ The CDDL (https://glassfish.java.net/public/CDDLv1.0.html)
 Apache Geode bundles the following file under the Common Development and
 Distribution License:

-  - javax.activation v1.1.1
     (http://java.sun.com/javase/technologies/desktop/javabeans/jaf/index.jsp),
-  - javax.mail v1.4.7 (http://www.oracle.com/)
-  - javax.resource v 1.7 (https://glassfish.java.net/)
+  - jaxb-istack-commons v2.2 (https://javaee.github.io/jaxb-istack-commons/)
+  - javax.activation v1.2.0
+    (https://www.oracle.com/technetwork/java/javase/jaf-135115.html)
+  - javax.mail v1.6.2 (http://www.oracle.com/)
+  - javax.resource v 1.7.1 (https://glassfish.java.net/)
   - javax.servlet v3.1.0 (https://glassfish.java.net/)
-  - javax.transaction v1.2 (https://glassfish.java.net/)
-  - javax.xml.bind v2.2.11
+  - javax.transaction v1.3 (https://glassfish.java.net/)
+  - jaxb v2.3.1 (https://javaee.github.io/jaxb-v2/)

 1. Definitions.
   1.1. Contributor means each individual or entity that creates or
@@ -602,6 +608,9 @@ The MIT License (http://opensource.org/licenses/mit-license.html)

 Apache Geode bundles the following files under the MIT license:

+  - Animal Sniffer Annotations v1.17
+    (https://www.mojohaus.org/animal-sniffer/source-repository.html)
+    Copyright (c) 2009 codehaus.org
   - JavaScript InfoVis Toolkit v2.0.1 (http://philogb.github.io/jit/),
     Copyright (c) 2011 Sencha Inc.
   - JOpt Simple (http://pholser.github.io/jopt-simple/), Copyright (c)
```

sboorlagadda commented:
@metatype Thanks for the feedback.
I see that `jaxb-istack-commons` has CDDL 1.1 (https://javaee.github.io/jaxb-istack-commons/LICENSE). So do we need to include it as 1.1 rather than 1.0?

metatype commented:
I also reviewed the jars for NOTICE(s) and pushed changes.

metatype commented:
> @metatype Thanks for the feedback.
> I see that `jaxb-istack-commons` has CDDL 1.1 (https://javaee.github.io/jaxb-istack-commons/LICENSE). So do we need to include it as 1.1 rather than 1.0?

yea makes sense

sboorlagadda commented:
@metatype Thanks. I have updated CDDL to 1.1

Also `Checker Framework` seems to be GPL but I see that you considered it as BSD. Also GPL is *[category x](https://www.apache.org/legal/resolved.html#category-x)* and not allowed.
https://github.com/typetools/checker-framework/blob/master/LICENSE.txt

metatype commented:
> @metatype Thanks. I have updated CDDL to 1.1
> 
> Also `Checker Framework` seems to be GPL but I see that you considered it as BSD. Also GPL is _[category x](https://www.apache.org/legal/resolved.html#category-x)_ and not allowed.
> https://github.com/typetools/checker-framework/blob/master/LICENSE.txt

Whoops, I meant to add checker-qual under the MIT section.  If you look at their license statement [1], the qual/* and related files are under the more permissive MIT license.  You can also follow the discussion of this in the guava community [2].

[1] https://github.com/typetools/checker-framework/blob/master/LICENSE.txt
[2] https://groups.google.com/forum/#!topic/guava-discuss/wUmhhOP2Kpo

sboorlagadda commented:
@metatype I did fix the license for checker-qual. Do you want to take a final look at the changes and give a 👍 ?

metatype commented:
LGTM
